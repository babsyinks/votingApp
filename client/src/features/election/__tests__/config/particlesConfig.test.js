import params from "../../config/particlesConfig";

describe("particlesConfig", () => {
  it("should match the expected shape and values snapshot", () => {
    expect(params).toMatchSnapshot();
  });

  it("should have essential top-level keys", () => {
    expect(params).toHaveProperty("fps_limit", 60);
    expect(params).toHaveProperty("interactivity");
    expect(params).toHaveProperty("particles");
    expect(params).toHaveProperty("retina_detect", true);
  });

  it("should define hover repulse mode", () => {
    expect(params.interactivity.events.onhover.mode).toBe("repulse");
  });

  it("should set particles color to white", () => {
    expect(params.particles.color.value).toBe("#ffffff");
  });
});
