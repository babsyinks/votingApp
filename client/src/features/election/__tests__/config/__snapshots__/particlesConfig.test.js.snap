// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`particlesConfig should match the expected shape and values snapshot 1`] = `
Object {
  "fps_limit": 60,
  "interactivity": Object {
    "detect_on": "canvas",
    "events": Object {
      "onclick": Object {
        "enable": true,
        "mode": "push",
      },
      "onhover": Object {
        "enable": true,
        "mode": "repulse",
        "parallax": Object {
          "enable": false,
          "force": 60,
          "smooth": 10,
        },
      },
      "resize": true,
    },
    "modes": Object {
      "attract": Object {
        "distance": 200,
        "duration": 0.4,
        "factor": 5,
      },
      "push": Object {
        "quantity": 4,
      },
    },
  },
  "particles": Object {
    "color": Object {
      "value": "#ffffff",
    },
    "line_linked": Object {
      "color": "#ffffff",
      "distance": 150,
      "enable": true,
      "opacity": 0.4,
      "width": 1,
    },
    "move": Object {
      "attract": Object {
        "enable": false,
        "rotateX": 600,
        "rotateY": 1200,
      },
      "bounce": false,
      "direction": "none",
      "enable": true,
      "out_mode": "out",
      "random": false,
      "speed": 2,
      "straight": false,
    },
    "number": Object {
      "density": Object {
        "enable": true,
        "value_area": 800,
      },
      "value": 80,
    },
    "opacity": Object {
      "anim": Object {
        "enable": false,
        "opacity_min": 0.1,
        "speed": 1,
        "sync": false,
      },
      "random": false,
      "value": 0.5,
    },
    "shape": Object {
      "character": Object {
        "fill": false,
        "font": "Verdana",
        "style": "",
        "value": "*",
        "weight": "400",
      },
      "image": Object {
        "height": 100,
        "replace_color": true,
        "src": "images/github.svg",
        "width": 100,
      },
      "polygon": Object {
        "nb_sides": 5,
      },
      "stroke": Object {
        "color": "#000000",
        "width": 0,
      },
      "type": "circle",
    },
    "size": Object {
      "anim": Object {
        "enable": false,
        "size_min": 0.1,
        "speed": 40,
        "sync": false,
      },
      "random": true,
      "value": 5,
    },
  },
  "polygon": Object {
    "draw": Object {
      "enable": false,
      "lineColor": "#ffffff",
      "lineWidth": 0.5,
    },
    "move": Object {
      "radius": 10,
    },
    "scale": 1,
    "type": "none",
    "url": "",
  },
  "retina_detect": true,
}
`;
